packages/amplify_core/lib/src/io/aws_file.dart

```diff
@@ -85,13 +82,14 @@ abstract class AWSFile {
   }) = AWSFilePlatform.fromStream;
 
   /// {@template amplify_core.io.aws_file.from_path}
-  /// Creates an [AWSFile] from a path, optionally specifying file name.
+  /// Creates an [AWSFile] from a path, optionally specifying a file [name].
   ///
-  /// In native, the [path] is the absolute path to a file in the file system.
+  /// On VM platforms, the [path] is the absolute path to a file in the file
+  /// system.
   ///
-  /// In Web, the [path] can be an object url created by
+  /// On Web, the [path] can be an object url created by
   /// [`URL.createObjectURL()`](https://developer.mozilla.org/en-US/docs/Web/API/URL/createObjectURL)
-  /// . When passing object url, it's recommended to set a value for [name].
+  /// . When passing an object url, it's recommended to set a value for [name].
   ///
   /// Throws [S3StorageException] if the file is not readable.
```

Comment:
Should this class throw an S3 specific error? Seems like we need a more general error

Reply:
Good catch, it should be the `InvalidFileException`. Updated.